File: Backend/spring_boot_backend_template/src/main/java/com/backend/services/ApplicationService.java
package com.backend.services;

import com.backend.entities.Application;
import com.backend.entities.ApplicationStatus;
import java.util.List;

public interface ApplicationService {
    Application createApplication(Long parentId, Long childId);
    Application getApplicationById(Long id);
    List<Application> getApplicationsByParentId(Long parentId);
    List<Application> getApplicationsByParent(Long parentId);
    List<Application> getAllApplications();
    List<Application> getApplicationsByStatus(ApplicationStatus status);
    Application updateApplicationStatus(Long id, ApplicationStatus status, String message);
    Application approveApplication(Long id, String message);
    Application rejectApplication(Long id, String message);
    Application requestDocuments(Long id, String message);
    void deleteApplication(Long id);
}

File: frontend/src/services/applicationService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/applications';

const getAuthHeader = () => ({
  headers: { Authorization: `Bearer ${localStorage.getItem('authToken')}` }
});

const applicationService = {
  createApplication: async (parentId, childId) => {
    const response = await axios.post(API_URL, { parentId, childId }, getAuthHeader());
    return response.data;
  },

  getApplicationById: async (id) => {
    const response = await axios.get(`${API_URL}/${id}`, getAuthHeader());
    return response.data;
  },

  getApplicationsByParent: async (parentId) => {
    const response = await axios.get(`${API_URL}/parent/${parentId}`, getAuthHeader());
    return response.data;
  },

  getAllApplications: async () => {
    const response = await axios.get(API_URL, getAuthHeader());
    return response.data;
  },

  getApplicationsByStatus: async (status) => {
    const response = await axios.get(`${API_URL}/status/${status}`, getAuthHeader());
    return response.data;
  },

  updateApplicationStatus: async (id, status, message) => {
    const response = await axios.put(`${API_URL}/${id}/status`, { status, message }, getAuthHeader());
    return response.data;
  },

  deleteApplication: async (id) => {
    const response = await axios.delete(`${API_URL}/${id}`, getAuthHeader());
    return response.data;
  }
};

export default applicationService;
